src/main/java/com/coravy/hudson/plugins/github/GithubLinkAction.java

```diff
@@ -30,4 +35,24 @@ public String getUrlName() {
         return projectProperty.getProjectUrl().baseUrl();
     }
 
+    @SuppressWarnings({"rawtypes"})
+    @Extension
+    public static class Factory extends TransientActionFactory<Job> {
+        @Override
+        public Class<Job> type() {
+            return Job.class;
+        }
+
+        @Override
+        public Collection<? extends Action> createFor(Job j) {
```

Comment:
Right, due to Java erasure it is not possible to do better, except perhaps with more exotic type tokens.